xcode/Subconscious/Shared/Components/Common/Profile/DidQrCodeView.swift

```diff
@@ -36,24 +37,80 @@ struct DidQrCodeView: View {
         let col = CIColor(red: r, green: g, blue: b, alpha: a)
         maskFilter.inputImage = CIImage(color: col)
         
-        guard let img = maskFilter.outputImage else { return nil }
-        guard let cgImg = context.createCGImage(img, from: img.extent) else { return nil }
+        guard let img = maskFilter.outputImage else {
+            return fallback
+        }
+        
+        let scaleFactor = 10.0
+        let upscaleImg = img
+            .samplingNearest()
+            .transformed(
+                by: CGAffineTransformMakeScale(scaleFactor, scaleFactor)
+            )
+        
+        guard let cgImg = context.createCGImage(
+            upscaleImg,
+            from: upscaleImg.extent
+        ) else {
+            return fallback
+        }
         
         return UIImage(cgImage: cgImg)
     }
+}
+
+struct DidQrCodeView: View {
+    var did: Did
+    var color: Color
     
+    var img: UIImage {
+        Self.generate(from: did.did, color: color)
+    }
     
     var body: some View {
-        let img = generateQRCode(from: "\(did.did)") ?? UIImage(systemName: "xmark.circle") ?? UIImage()
-        
-        Image(uiImage: img)
+        DidQrCodeImage(image: img)
+    }
+}
+
+struct DidQrCodeImage: View {
+    var image: UIImage
+    
+    var body: some View {
+        Image(uiImage: image)
             .resizable()
             .interpolation(.none)
             .scaledToFit()
             .padding(AppTheme.unit2)
     }
 }
 
+struct ShareableDidQrCodeView: View {
+    var did: Did
+    var color: Color
+    
+    var img: UIImage {
+        DidQrCodeView.generate(from: did.did, color: color)
+    }
+    
+    var body: some View {
+        let shareable = Image(uiImage: img)
```

Comment:
This is the easiest way I found to get an imagine into a `Transferable` protocol-compatible form.